### DISCUSSION TITLE
Can I index samba shared files (NAS directory) without moving them?

### DISCUSSION THREAD

**Discussion by emanueletocci** [immich-app/immich#8041] [General]
Hi, I'm looking for a self hosted photo gallery to index all my photos and Immich seem to be the best project to do this. But I have a problem: all my phots are actually stored on NAS (trueNAS) and shared through samba or NFS. Is possibile to index these files without moving them to another path?

**Comment by JW-CH** [comment]
Take a look at this: https://immich.app/docs/features/libraries#external-libraries

**Reply by emanueletocci** [reply]
> Take a look at this: https://immich.app/docs/features/libraries#external-libraries

Thank you
